platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java

```diff
@@ -319,7 +309,10 @@ public long getClassId() {
 
     @Override
     public int getVersion() {
-        return ClassVersion.SOFTWARE_VERSION;
+        if (deserializedVersion < ClassVersion.ADDRESS_BOOK_ROUND) {
+            return deserializedVersion;
+        }
+        return ClassVersion.ADDRESS_BOOK_ROUND;
```

Comment:
why not just return `deserializedVersion`?

Reply:
fair enough, I will make that change.